src/utils/service_builder.rs

```diff
@@ -207,11 +209,13 @@ fn build_backend_handlers(
 fn build_providers(
     configs: &[ProviderConfig],
     kim_factorys: HashMap<String, KeyInfoManagerFactory>,
-) -> Result<Vec<(ProviderId, Provider)>> {
+) -> Result<Vec<(ProviderId, Provider, String)>> {
     let mut list = Vec::new();
+    let mut provider_names = Vec::new();
```

Comment:
Nerd comment 🤓 : this could be a `HashSet` to make `contains` `O(1)` instead of `O(n)`
Note that you can ignore since `n`, the number of provider will likely not be `u32::MAX`

Reply:
Very true. Good spot 🤓

Reply:
I'm genuinely curios if for cases like this (very small sets) there is actually a difference in speed. It's probably insignificant, but I guess for neatness' sake `HashSet` is better.